qemu/tests/blockdev_commit_general_operation.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from virttest import utils_misc
+
+from provider import job_utils
+from provider import backup_utils
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitGeneralOperation(BlockDevCommitTest):
+
+    def commit_op(self, cmd, args=None):
+        self.main_vm.monitor.cmd(cmd, args)
+        job_status = job_utils.query_block_jobs(self.main_vm)
+        logging.info("The status after operation %s is %s", cmd, job_status)
+
+    def commit_snapshots(self):
+        device = self.params.get("device_tag")
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        options = ["speed", "auto-finalize", "auto-dismiss"]
+        arguments = self.params.copy_from_keys(options)
+        arguments["speed"] = self.params["commit_speed"]
+        arguments["auto-finalize"] = self.params.get_boolean("finalize")
+        arguments["auto-dismiss"] = self.params.get_boolean("dismiss")
+        device = self.get_node_name(snapshot_tags[-1])
+        commit_cmd = backup_utils.block_commit_qmp_cmd
+        cmd, args = commit_cmd(device, **arguments)
+        self.main_vm.monitor.cmd(cmd, args)
+        job_id = args.get("job-id", device)
+        self.main_vm.monitor.cmd("block-job-set-speed",
+                                 {'device': job_id, 'speed': 10240})
+        self.commit_op("stop")
+        self.commit_op("cont")
+        self.commit_op("job-pause", {'id': job_id})
+        self.commit_op("job-resume", {'id': job_id})
+        self.commit_op("job-cancel", {'id': job_id})
+        utils_misc.wait_for(lambda: not self.main_vm.get_job_status(device),
```

Comment:
event = job_utils.get_event_by_condition(
             self.main_vm, 'BLOCK_JOB_CANCELLED',
             self.params.get_numeric('job_cancelled_timeout', 60),
             device=job_id
         )
         if event is None:
             self.test.fail('Job failed to cancel')
If we are sure the job should be cancelled without any error, there must  be a 'BLOCK_JOB_CANCELLED' event

Reply:
Okay,thanks.